packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -0,0 +1,190 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'notification_listener.dart';
+import 'scroll_controller.dart';
+import 'scroll_notification.dart';
+import 'scroll_physics.dart';
+import 'scroll_position.dart';
+import 'scroll_view.dart';
+import 'sliver.dart';
+
+class PageController extends ScrollController {
+  PageController({
+    this.initialPage: 0.0,
+  });
+
+  final double initialPage;
+
+  double get page {
+    final ScrollPosition position = this.position;
+    return position.pixels / position.viewportDimension;
+  }
+
+  Future<Null> animateToPage(int page, {
+    @required Duration duration,
+    @required Curve curve,
```

Comment:
We do that in `AnimationController`, but we don't do that in `ScrollController` (today) because that would look ugly...  Not sure what the best thing to do is.